sdk/communication/communication-identity/test/public/utils/recordedClient.ts

```diff
@@ -48,107 +35,91 @@ const replaceableVariables: { [k: string]: string } = {
-  replaceableVariables,
-  customizationsOnRecordings: [
-    (recording: string): string =>
-      recording.replace(/"token"\s?:\s?"[^"]*"/g, `"token":"sanitized"`),
-    (recording: string): string =>
-      recording.replace(/"access_token"\s?:\s?"[^"]*"/g, `"access_token":"sanitized"`),
-    (recording: string): string => recording.replace(/(https:\/\/)([^/',]*)/, "$1endpoint"),
-    (recording: string): string => recording.replace(/"id"\s?:\s?"[^"]*"/g, `"id":"sanitized"`),
-    (recording: string): string => {
-      return recording.replace(
-        /(https:\/\/[^/',]*\/identities\/)[^/',]*(\/token)/,
-        "$1sanitized$2"
-      );
+const sanitizerOptions: SanitizerOptions = {
+  connectionStringSanitizers: [
+    {
+      actualConnString: env.COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING,
+      fakeConnString: envSetupForPlayback["COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING"],
+    },
+  ],
+  uriSanitizers: [
+    {
+      regex: true,
+      target: `(.*)/identities/(?<secret_content>.*?)[/|?](.*)`,
+      value: "sanitized",
+      groupForReplace: "secret_content",
     },
-    (recording: string): string =>
-      recording.replace(/\/identities\/[^/'",]*/, "/identities/sanitized"),
-    (recording: string): string => recording.replace(/\+\d{1}\d{3}\d{3}\d{4}/g, "+18005551234"),
-    (recording: string): string =>
-      recording.replace(
-        /[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}/gi,
-        "00000000-0000-0000-0000-000000000000"
-      ),
   ],
-  queryParametersToSkip: [],
+  generalSanitizers: [
+    { regex: true, target: `"token"\\s?:\\s?"[^"]*"`, value: `"token":"sanitized"` },
+    { regex: true, target: `"id"\\s?:\\s?"[^"]*"`, value: `"id":"sanitized"` },
+    {
+      regex: true,
+      target: `[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}`,
+      value: `sanitized`,
+    },
+  ],
+};
+
+const recorderOptions: RecorderStartOptions = {
+  envSetupForPlayback,
+  sanitizerOptions: sanitizerOptions,
 };
 
-export function createRecordedCommunicationIdentityClient(
+export async function createRecorder(context: Test | undefined): Promise<Recorder> {
+  const recorder = new Recorder(context);
+  await recorder.start(recorderOptions);
+  await recorder.setMatcher("CustomDefaultMatcher", {
+    excludedHeaders: [
+      "Accept-Language", // This is env-dependent
+      "x-ms-content-sha256", // This is dependent on the current datetime
+    ],
+  });
+  return recorder;
+}
+
+export async function createRecordedCommunicationIdentityClient(
   context: Context
-): RecordedClient<CommunicationIdentityClient> {
-  const recorder = record(context, environmentSetup);
+): Promise<RecordedClient<CommunicationIdentityClient>> {
+  const recorder = await createRecorder(context.currentTest);
+
+  const client = new CommunicationIdentityClient(
+    env.COMMUNICATION_LIVETEST_DYNAMIC_CONNECTION_STRING ?? "",
+    recorder.configureClientOptions({})
+  );
 
   // casting is a workaround to enable min-max testing
```

Comment:
nit: this appears to be an outdated comment

Reply:
thanks! removed